Code Audit Routines — Open: 1 | Closed: 1192 | Simplified: | Metric | Count |

## Code Audit Routines

**Last sweep**: `2026-04-29T08:51:02Z`
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5
**Badge status**: 

### Quality

| Metric | Value |
| --- | --- |
| SonarCloud gate | ERROR |
| SonarCloud issues | 64 (0 BLOCKER · 0 CRITICAL · 64 MAJOR · 0 MINOR · 0 INFO) |
| Qlty grade (local, from smell count) | UNKNOWN |
| Qlty smells | 0 → 0 (unchanged) |

### Simplification

| Metric | Value |
| --- | --- |
| Open | 1 |
| Closed | 1192 |
| Simplified (state tracked) | | Metric | Count | |
| Resolution rate | 99% |

### PR Scanning

| Metric | Value |
| --- | --- |
| PRs scanned (lifetime) | 2993 |
| Issues created (lifetime) | 2261 |

### Bot Review Coverage

---
_Auto-updated by daily quality sweep. Findings as of `2026-04-29T08:51:02Z` — may not reflect recent merges between sweeps._
_The codebase is the primary source of truth. This dashboard is a reporting snapshot that may lag behind reality._
_Do not edit manually._

## Daily Code Quality Sweep

**Date**: 2026-03-05T01:59:17Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 2

**Top findings:**
  - `python-env/dspy-env/lib/python3.9/site-packages/tqdm/completion.sh`: Prefer mapfile or read -a to split command output (or quote to avoid splitting). [SC2207]
  - `python-env/dspy-env/lib/python3.9/site-packages/tqdm/completion.sh`: Double quote to prevent globbing and word splitting. [SC2086]
  - `python-env/dspy-env/lib/python3.9/site-packages/tqdm/completion.sh`: Prefer mapfile or read -a to split command output (or quote to avoid splitting). [SC2207]
  - `tests/test-ai-actions.sh`: ../.agents/scripts/supervisor/ai-reason.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-container-pool.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-container-pool.sh`: ./database.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-container-pool.sh`: ./container-pool.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-contest-helper.sh`: ../.agents/scripts/contest-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-backup-safety.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-ai-supervisor-e2e.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-ai-supervisor-e2e.sh`: ./ai-context.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-ai-supervisor-e2e.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/seo-export-ahrefs.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/seo-export-ahrefs.sh`: ./credentials.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/github-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/readme-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/setup-linters-wizard.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/sonarcloud-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/onboarding-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/linter-manager.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/document-creation-helper.sh`: ./bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/todo-ready.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/stagehand-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/response-scoring-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/stagehand-setup.sh`: ./../../.agents/scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/entity-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/webhosting-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/email-signature-parser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/quality-cli-manager.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/postflight-check.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/thumbnail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/generate-claude-commands.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/servers-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/sonarscanner-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/quickfile-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/backfill-blocked-comments.sh`: ./issue-sync.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/backfill-blocked-comments.sh`: ./../database.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/backfill-blocked-comments.sh`: ./../../supervisor-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/ai-context.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/supervisor-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/supervisor-helper.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/supervisor-helper.sh`: ./database.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/ai-reason.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/ai-reason.sh`: ai-context.sh was not specified as input (see shellcheck -x). [SC1091]

### Qlty Maintainability Smells

_No smells detected or qlty analysis returned empty._

### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 107
- **By severity**:
  - MAJOR: 74
  - MINOR: 33
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 107
  - BUG: 0
  - VULNERABILITY: 0


### CodeRabbit

_Monitoring: 107 issues (delta: 107), gate OK, 74 high/critical — no active review needed. (baseline established — deltas will be computed from next sweep)_

### Merged PR Review Scanner

- **PRs scanned**: 10
- **Findings**: 42
- **Issues created**: 13

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._


---
_Auto-generated by pulse-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-03-11T22:15:02Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 4

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 7

**Top findings:**
  - `tests/test-headless-runtime-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `tests/test-headless-runtime-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `tests/test-memory-consolidation.sh`: ./.agents/scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-memory-consolidation.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `tests/test-memory-consolidation.sh`: AUDIT_MARKER appears unused. Verify use (or export if used externally). [SC2034]
  - `tests/test-ai-supervisor-e2e.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-ai-supervisor-e2e.sh`: ./ai-context.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-ai-supervisor-e2e.sh`: PATH was modified in a subshell. That change might be lost. [SC2031]
  - `tests/test-container-pool.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-container-pool.sh`: ./database.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-container-pool.sh`: ./container-pool.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-contest-helper.sh`: ../.agents/scripts/contest-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-backup-safety.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-backup-safety.sh`: Use find instead of ls to better handle non-alphanumeric filenames. [SC2012]
  - `tests/test-backup-safety.sh`: Use find instead of ls to better handle non-alphanumeric filenames. [SC2012]
  - `tests/test-supervisor-state-machine.sh`: ../.agents/scripts/supervisor-archived/sanity-check.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-ai-actions.sh`: ../.agents/scripts/supervisor/ai-reason.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/stash-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/session-count-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/issue-sync-lib.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/privacy-filter-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/process-guard-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/process-guard-helper.sh`: Consider using pgrep instead of grepping ps output. [SC2009]
  - `.agents/scripts/process-guard-helper.sh`: Consider using pgrep instead of grepping ps output. [SC2009]
  - `.agents/scripts/setup-linters-wizard.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/model-availability-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/stats-wrapper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/stats-wrapper.sh`: ./worker-lifecycle-common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/stats-wrapper.sh`: LOGFILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/supervisor-archived/ai-context.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/scripts/supervisor-archived/ai-context.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/scripts/supervisor-archived/ai-context.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/deploy.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/supervisor-archived/ai-reason.sh`: ./../shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/ai-reason.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/ai-reason.sh`: ai-context.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/supervisor-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/supervisor-helper.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/supervisor-helper.sh`: ./database.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/issue-sync.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/supervisor-archived/cron.sh`: launchd.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/database.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/database.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/release.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/pulse.sh`: out of memory
  - `.agents/scripts/supervisor-archived/ai-actions.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/ai-actions.sh`: ai-context.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/ai-actions.sh`: ai-reason.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/cleanup.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/_common.sh`: ./../shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/_common.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/_common.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/utility.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/supervisor-archived/utility.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/supervisor-archived/utility.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/supervisor-archived/backfill-blocked-comments.sh`: ./issue-sync.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/backfill-blocked-comments.sh`: ./../database.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/backfill-blocked-comments.sh`: ./../../supervisor-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/dispatch.sh`: ../worker-sandbox-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/dispatch.sh`: ../verify-operation-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/dispatch.sh`: ../worker-sandbox-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/list-keys-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/verify-run-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/gsc-sitemap-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/secretlint-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/vaultwarden-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/yt-dlp-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/higgsfield-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/toon-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/thumbnail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/worker-token-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/worker-token-helper.sh`: token_info appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/worker-token-helper.sh`: repo_check appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]


### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 402
- **By severity**:
  - MINOR: 309
  - MAJOR: 93
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 402
  - BUG: 0
  - VULNERABILITY: 0


### CodeRabbit

_Monitoring: 402 issues (delta: 6), gate OK — no active review needed._

### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 30
- **Issues created**: 10

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._


---
_Auto-generated by pulse-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-03-12T22:30:02Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 4

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 5

**Top findings:**
  - `tests/test-headless-runtime-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `tests/test-headless-runtime-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `tests/test-memory-consolidation.sh`: ./.agents/scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-memory-consolidation.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `tests/test-memory-consolidation.sh`: AUDIT_MARKER appears unused. Verify use (or export if used externally). [SC2034]
  - `tests/test-ai-supervisor-e2e.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-ai-supervisor-e2e.sh`: ./ai-context.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-ai-supervisor-e2e.sh`: PATH was modified in a subshell. That change might be lost. [SC2031]
  - `tests/test-container-pool.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-container-pool.sh`: ./database.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-container-pool.sh`: ./container-pool.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-contest-helper.sh`: ../.agents/scripts/contest-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-backup-safety.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-backup-safety.sh`: Use find instead of ls to better handle non-alphanumeric filenames. [SC2012]
  - `tests/test-backup-safety.sh`: Use find instead of ls to better handle non-alphanumeric filenames. [SC2012]
  - `tests/test-supervisor-state-machine.sh`: ../.agents/scripts/supervisor-archived/sanity-check.sh was not specified as input (see shellcheck -x). [SC1091]
  - `tests/test-ai-actions.sh`: ../.agents/scripts/supervisor/ai-reason.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/stash-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/session-count-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/issue-sync-lib.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/privacy-filter-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/process-guard-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/process-guard-helper.sh`: Consider using pgrep instead of grepping ps output. [SC2009]
  - `.agents/scripts/process-guard-helper.sh`: Consider using pgrep instead of grepping ps output. [SC2009]
  - `.agents/scripts/setup-linters-wizard.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/model-availability-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/stats-wrapper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/stats-wrapper.sh`: ./worker-lifecycle-common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/stats-wrapper.sh`: LOGFILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/supervisor-archived/ai-context.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/scripts/supervisor-archived/ai-context.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/scripts/supervisor-archived/ai-context.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/deploy.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/supervisor-archived/ai-reason.sh`: ./../shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/ai-reason.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/ai-reason.sh`: ai-context.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/supervisor-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/supervisor-helper.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/supervisor-helper.sh`: ./database.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/issue-sync.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/supervisor-archived/cron.sh`: launchd.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/database.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/database.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/release.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/pulse.sh`: out of memory
  - `.agents/scripts/supervisor-archived/ai-actions.sh`: ./_common.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/ai-actions.sh`: ai-context.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/ai-actions.sh`: ai-reason.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/cleanup.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/_common.sh`: ./../shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/_common.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/_common.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/supervisor-archived/utility.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/supervisor-archived/utility.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/supervisor-archived/utility.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/supervisor-archived/backfill-blocked-comments.sh`: ./issue-sync.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/backfill-blocked-comments.sh`: ./../database.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/backfill-blocked-comments.sh`: ./../../supervisor-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/dispatch.sh`: ../worker-sandbox-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/dispatch.sh`: ../verify-operation-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/supervisor-archived/dispatch.sh`: ../worker-sandbox-helper.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/list-keys-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/verify-run-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/gsc-sitemap-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/secretlint-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/vaultwarden-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/yt-dlp-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/higgsfield-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]


### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 404
- **By severity**:
  - MINOR: 311
  - MAJOR: 93
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 404
  - BUG: 0
  - VULNERABILITY: 0


### CodeRabbit

_Monitoring: 404 issues (delta: 2), gate OK — no active review needed._

### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 54
- **Issues created**: 25

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._


---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-03-29T20:00:01Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)


  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-03-30T06:08:27Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)


  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-05T21:45:02Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)


  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-07T01:43:43Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)


  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-10T02:15:01Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)


  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-10T22:58:35Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)


  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-11T02:32:39Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)


  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-11T23:14:33Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)


  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-12T02:50:57Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)


  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-12T23:29:38Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)


  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-13T03:09:44Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)


  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-13T23:41:33Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 46
- **Notes**: 102

**Top findings:**
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/dev-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dns-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/document-creation-helper.sh`: ./bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/document-creation-helper.sh`: to_ext_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/document-creation-helper.sh`: force_tool_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/document-extraction-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/domain-research-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/draft-response-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dspy-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dspy-helper.sh`: ./bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dspyground-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._

- **Qlty Cloud grade**: UNKNOWN
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 226
- **By severity**:
  - MINOR: 161
  - MAJOR: 65
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 226
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 161 issues
  - shelldre:S1066: 65 issues

### CodeRabbit

_Monitoring: 226 issues (delta: 1), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-14T03:16:48Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 45
- **Notes**: 97

**Top findings:**
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/dev-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dns-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/document-creation-helper.sh`: ./bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/document-creation-helper.sh`: to_ext_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/document-creation-helper.sh`: force_tool_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/document-extraction-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/domain-research-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/draft-response-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dspy-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dspy-helper.sh`: ./bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dspyground-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._

- **Qlty Cloud grade**: UNKNOWN
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 231
- **By severity**:
  - MINOR: 165
  - MAJOR: 66
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 231
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 165 issues
  - shelldre:S1066: 66 issues

### CodeRabbit

_Monitoring: 231 issues (delta: 5), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 3
- **Issues created**: 0

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-14T23:56:33Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 45
- **Notes**: 99

**Top findings:**
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/dev-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dns-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/document-creation-helper.sh`: ./bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/document-creation-helper.sh`: to_ext_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/document-creation-helper.sh`: force_tool_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/document-extraction-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 235
- **By severity**:
  - MINOR: 166
  - MAJOR: 69
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 235
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 166 issues
  - shelldre:S1066: 69 issues

### CodeRabbit

_Monitoring: 235 issues (delta: 9), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-15T03:23:10Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 45
- **Notes**: 97

**Top findings:**
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/dev-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dns-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/document-creation-helper.sh`: ./bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/document-creation-helper.sh`: to_ext_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/document-creation-helper.sh`: force_tool_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/document-extraction-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 235
- **By severity**:
  - MINOR: 166
  - MAJOR: 69
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 235
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 166 issues
  - shelldre:S1066: 69 issues

### CodeRabbit

_Monitoring: 235 issues (delta: 4), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-16T00:11:33Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 27
- **Notes**: 105

**Top findings:**
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/deploy-agents-on-merge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/dev-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dns-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/doctor-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 238
- **By severity**:
  - MINOR: 166
  - MAJOR: 72
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 238
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 166 issues
  - shelldre:S1066: 72 issues

### CodeRabbit

_Monitoring: 238 issues (delta: 3), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-16T03:29:02Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 27
- **Notes**: 103

**Top findings:**
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/deploy-agents-on-merge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/dev-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dns-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/doctor-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 239
- **By severity**:
  - MINOR: 166
  - MAJOR: 73
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 239
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 166 issues
  - shelldre:S1066: 73 issues

### CodeRabbit

_Monitoring: 239 issues (delta: 4), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 17
- **Issues created**: 0

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-17T00:24:35Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 27
- **Notes**: 105

**Top findings:**
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/deploy-agents-on-merge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/dev-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dns-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 238
- **By severity**:
  - MINOR: 166
  - MAJOR: 72
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 238
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 166 issues
  - shelldre:S1066: 72 issues

### CodeRabbit

_Monitoring: 238 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-18T00:34:37Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 27
- **Notes**: 105

**Top findings:**
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/deploy-agents-on-merge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/dev-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 238
- **By severity**:
  - MINOR: 165
  - MAJOR: 73
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 238
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 165 issues
  - shelldre:S1066: 73 issues

### CodeRabbit

_Monitoring: 238 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-19T00:43:45Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 27
- **Notes**: 105

**Top findings:**
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/deploy-agents-on-merge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/dev-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 251
- **By severity**:
  - MINOR: 179
  - MAJOR: 72
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 251
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 167 issues
  - shelldre:S1066: 71 issues
  - shelldre:S100: 12 issues
  - shell:S6573: 1 issues

### CodeRabbit

**Trigger**: issue spike +13

@coderabbitai Please run a full codebase review of this repository. Focus on:
- Security vulnerabilities and credential exposure
- Shell script quality (error handling, quoting, race conditions)
- Code duplication and maintainability
- Documentation accuracy
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._

## Daily Code Quality Sweep

**Date**: 2026-04-20T00:53:34Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 28
- **Notes**: 109

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/deploy-agents-on-merge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/dev-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 270
- **By severity**:
  - MINOR: 188
  - MAJOR: 82
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 270
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 171 issues
  - shelldre:S1066: 81 issues
  - shelldre:S100: 17 issues
  - shell:S6573: 1 issues

### CodeRabbit

**Trigger**: issue spike +19

@coderabbitai Please run a full codebase review of this repository. Focus on:
- Security vulnerabilities and credential exposure
- Shell script quality (error handling, quoting, race conditions)
- Code duplication and maintainability
- Documentation accuracy
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.8.78 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-04-20T01:21:19Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 28
- **Notes**: 106

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/deploy-agents-on-merge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/dev-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/dispatch-dedup-footprint.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
### Qlty Maintainability

- **Total smells**: 23
- **Grade (local, from count)**: B
- **By rule (fix these for maximum grade improvement)**:
  - qlty:file-complexity: 23
- **Top files (highest smell density)**:
  - .agents/plugins/opencode-aidevops/oauth-pool-auth.mjs: 1 smells
  - .agents/plugins/opencode-aidevops/oauth-pool-callback.mjs: 1 smells
  - .agents/plugins/opencode-aidevops/oauth-pool-display.mjs: 1 smells
  - .agents/plugins/opencode-aidevops/provider-auth-request.mjs: 1 smells
  - .agents/scripts/add-related-docs.py: 1 smells
  - .agents/scripts/cross-document-linking.py: 1 smells
  - .agents/scripts/email-voice-miner.py: 1 smells
  - .agents/scripts/email_jmap_sync.py: 1 smells
  - .agents/scripts/email_normaliser_sections.py: 1 smells
  - .agents/scripts/email_summary_heuristic.py: 1 smells
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 270
- **By severity**:
  - MINOR: 188
  - MAJOR: 82
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 270
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 171 issues
  - shelldre:S1066: 81 issues
  - shelldre:S100: 17 issues
  - shell:S6573: 1 issues

### CodeRabbit

**Trigger**: issue spike +76

@coderabbitai Please run a full codebase review of this repository. Focus on:
- Security vulnerabilities and credential exposure
- Shell script quality (error handling, quoting, race conditions)
- Code duplication and maintainability
- Documentation accuracy
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.8.78 plugin for [OpenCode](https://opencode.ai) v1.14.18 with claude-opus-4-7 spent 10m and 21,762 tokens on this as a headless worker.

## Daily Code Quality Sweep

**Date**: 2026-04-21T00:59:34Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 29
- **Notes**: 110

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/backfill-orphan-todos.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dashboard-freshness-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/deploy-agents-on-merge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/design-preview-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 288
- **By severity**:
  - MINOR: 196
  - MAJOR: 92
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 288
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 178 issues
  - shelldre:S1066: 91 issues
  - shelldre:S100: 18 issues
  - shell:S6573: 1 issues

### CodeRabbit

**Trigger**: issue spike +18

@coderabbitai Please run a full codebase review of this repository. Focus on:
- Security vulnerabilities and credential exposure
- Shell script quality (error handling, quoting, race conditions)
- Code duplication and maintainability
- Documentation accuracy
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 0
- **Issues created**: 0

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.8.86 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-04-21T01:35:35Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 29
- **Notes**: 106

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dashboard-freshness-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/deploy-agents-on-merge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/design-preview-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/detect-app-type.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 288
- **By severity**:
  - MINOR: 196
  - MAJOR: 92
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 288
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 178 issues
  - shelldre:S1066: 91 issues
  - shelldre:S100: 18 issues
  - shell:S6573: 1 issues

### CodeRabbit

**Trigger**: issue spike +18

@coderabbitai Please run a full codebase review of this repository. Focus on:
- Security vulnerabilities and credential exposure
- Shell script quality (error handling, quoting, race conditions)
- Code duplication and maintainability
- Documentation accuracy
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 44
- **Issues created**: 12

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.8.87 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-04-22T02:09:38Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 30
- **Notes**: 105

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: platform-detect.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: Did you forget to close this double quoted string? [SC1078]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-activity-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-dispatch.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dashboard-freshness-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/dataset-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 294
- **By severity**:
  - MINOR: 195
  - MAJOR: 99
  - INFO: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 294
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shelldre:S1481: 177 issues
  - shelldre:S1066: 98 issues
  - shelldre:S100: 18 issues
  - shell:S6573: 1 issues

### CodeRabbit

_Monitoring: 294 issues (delta: 6), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 12
- **Issues created**: 4

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.8.92 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-04-23T03:17:26Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 63
- **Notes**: 101

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_judge appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_max_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_dry_run appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_prompt appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_models appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_workdir appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/compare-models-scoring-lib.sh`: found_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: source_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: active_key_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-activity-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/credential-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crewai-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 1
- **By severity**:
  - MAJOR: 1
  - INFO: 0
  - MINOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 1
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shell:S6573: 1 issues

### CodeRabbit

_Monitoring: 1 issues (delta: -293), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 10
- **Issues created**: 4

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.8.95 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-04-24T03:50:56Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 63
- **Notes**: 103

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: pulse-cleanup.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_judge appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_max_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_dry_run appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_prompt appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_models appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_workdir appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/compare-models-scoring-lib.sh`: found_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: source_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: active_key_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-activity-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/counter-stack-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/crawl4ai-examples.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 1
- **By severity**:
  - MAJOR: 1
  - INFO: 0
  - MINOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 1
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shell:S6573: 1 issues

### CodeRabbit

_Monitoring: 1 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 5
- **Findings**: 12
- **Issues created**: 5

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.10.0 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-04-25T05:02:31Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 63
- **Notes**: 102

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: pulse-cleanup.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_judge appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_max_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_dry_run appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_prompt appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_models appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_workdir appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/compare-models-scoring-lib.sh`: found_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: source_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: active_key_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-activity-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/counter-stack-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 1
- **By severity**:
  - MAJOR: 1
  - INFO: 0
  - MINOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 1
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shell:S6573: 1 issues

### CodeRabbit

_Monitoring: 1 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 12
- **Issues created**: 4

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.11.0 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-04-26T06:16:39Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 63
- **Notes**: 102

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: pulse-cleanup.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_judge appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_max_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_dry_run appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_prompt appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_models appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_workdir appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/compare-models-scoring-lib.sh`: found_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: source_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: active_key_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contribution-watch-helper.sh`: POLL_DORMANT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/contribution-watch-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/contributor-activity-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contributor-insight-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/conversation-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/coolify-cli-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coolify-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/counter-stack-check.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 1
- **By severity**:
  - MAJOR: 1
  - INFO: 0
  - MINOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 1
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shell:S6573: 1 issues

### CodeRabbit

_Monitoring: 1 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 4
- **Findings**: 15
- **Issues created**: 5

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.11.7 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-04-27T07:18:08Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 60
- **Notes**: 94

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: pulse-cleanup.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_judge appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_max_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_dry_run appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_prompt appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_models appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_workdir appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/compare-models-scoring-lib.sh`: found_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: source_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: active_key_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-fanout-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/content-scanner-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/contest-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/context-builder-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 1
- **By severity**:
  - MAJOR: 1
  - INFO: 0
  - MINOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 1
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shell:S6573: 1 issues

### CodeRabbit

_Monitoring: 1 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 30
- **Findings**: 8
- **Issues created**: 1

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.13.0 automated scan.

## Daily Code Quality Sweep

**Date**: 2026-04-28T07:47:29Z
**Repo**: `marcusquinn/aidevops`
**Tools run**: 5

---

### ShellCheck (100 files scanned)

- **Errors**: 0
- **Warnings**: 58
- **Notes**: 89

**Top findings:**
  - `.agents/custom/scripts/r-stub-title-scan.sh`: SCRIPT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: ../../scripts/shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/custom/scripts/r-stub-title-scan.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/hooks/credential-emission-pre-push.sh`: $/${} is unnecessary on arithmetic variables. [SC2004]
  - `.agents/hooks/privacy-guard-pre-push.sh`: local_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/privacy-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/hooks/scope-guard-pre-push.sh`: remote_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/101domains-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/accessibility-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/add-skill-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-browser-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-sources-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agent-test-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/agno-setup.sh`: venv/bin/activate was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-cli-config.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-judgment-helper.sh`: FALLBACK_IDLE_TIMEOUT appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-judgment-helper.sh`: JUDGMENT_CACHE_TTL appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-research-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-research-helper.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/ai-threshold-judge.sh`: LOG_PREFIX appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ai-threshold-judge.sh`: last_messages appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/aidevops-update-check.sh`: lib/version.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/aidevops-update-check.sh`: SCRIPT_DIR may not be assigned. Did you mean script_dir? [SC2153]
  - `.agents/scripts/aidevops-update-check.sh`: git_context_val appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/ampcode-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/apple-mail-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/approval-helper.sh`: APPROVAL_KEY appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/attribution-detection-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/attribution-detection-helper.sh`: Don't use variables in the printf format string. Use printf '..%s..' "$foo". [SC2059]
  - `.agents/scripts/audit-agent-deployment.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-agent-deployment.sh`: CLAUDE_AGENT_DIR appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/audit-log-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/audit-log-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/auto-decomposer-scanner.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_SKILL_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-update-helper.sh`: DEFAULT_OPENCLAW_FRESHNESS_HOURS appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/auto-version-bump.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/autoagent-metric-helper.sh`: baseline_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/autogen-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-origin-labels.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/backfill-orphan-todos.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/badges-check-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/bash-upgrade-helper.sh`: Expressions don't expand in single quotes, use double quotes for that. [SC2016]
  - `.agents/scripts/batch-strategy-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/beads-sync-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/brief-tier-test-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/brief-tier-test-helper.sh`: timestamp appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/browser-qa-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/browser-qa-worker.sh`: base_url appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/budget-analysis-helper.sh`: ROUTING_TABLE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/budget-analysis-helper.sh`: See if you can use ${variable//search/replace} instead. [SC2001]
  - `.agents/scripts/bundle-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/campaigns-provision-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-draft-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/case-helper.sh`: Note that A && B || C is not if-then-else. C may run when A is true. [SC2015]
  - `.agents/scripts/check-workflows-helper.sh`: This function is never invoked. Check usage (or ignored if invoked indirectly). [SC2329]
  - `.agents/scripts/chrome-webstore-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/circuit-breaker-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id-counter.sh`: bootstrap_result appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/claim-task-id-issue.sh`: REPO_PATH may not be assigned. Did you mean repo_path? [SC2153]
  - `.agents/scripts/claim-task-id.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/claim-task-id.sh`: COUNTER_FILE appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/clawdhub-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cleanup-worktrees-async-helper.sh`: pulse-cleanup.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/cloudron-package-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli-chunked.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-cli.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/codacy-collector-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/code-audit-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-cli.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/coderabbit-pro-analysis.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/colormind-helper.sh`: SCRIPT_NAME appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/colormind-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/comfy-cli-helper.sh`: RED appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: GREEN appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comfy-cli-helper.sh`: YELLOW appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_judge appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_max_tokens appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-bench-lib.sh`: args_dry_run appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_prompt appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_models appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-cross-review-lib.sh`: cr_workdir appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/compare-models-scoring-lib.sh`: found_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: source_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/compare-models-scoring-lib.sh`: active_key_ref appears unused. Verify use (or export if used externally). [SC2034]
  - `.agents/scripts/comprehension-benchmark-helper.sh`: shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/config-helper.sh`: Command appears to be unreachable. Check usage (or ignore if invoked indirectly). [SC2317]
  - `.agents/scripts/contacts-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-calendar-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
  - `.agents/scripts/content-classifier-helper.sh`: ./shared-constants.sh was not specified as input (see shellcheck -x). [SC1091]
### Qlty Maintainability

_Qlty analysis returned empty or failed to parse._
### SonarCloud Quality Gate

- **Status**: OK
- **new_reliability_rating**: 1 (OK)
- **new_security_rating**: 1 (OK)
- **new_maintainability_rating**: 1 (OK)
- **new_duplicated_lines_density**: 0.0 (OK)
- **new_security_hotspots_reviewed**: 100.0 (OK)

- **Open issues**: 1
- **By severity**:
  - MAJOR: 1
  - INFO: 0
  - MINOR: 0
  - CRITICAL: 0
  - BLOCKER: 0
- **By type**:
  - CODE_SMELL: 1
  - BUG: 0
  - VULNERABILITY: 0

- **Top rules (fix these for maximum badge improvement)**:
  - shell:S6573: 1 issues

### CodeRabbit

_Monitoring: 1 issues (delta: 0), gate OK — no active review needed._
### Merged PR Review Scanner

- **PRs scanned**: 19
- **Findings**: 21
- **Issues created**: 12

_Issues labelled `quality-debt` — capped at 30% of dispatch concurrency._

---
_Auto-generated by stats-wrapper.sh daily quality sweep. The supervisor will review findings and create actionable issues._
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.13.5 automated scan.

--- a/.agents/scripts/cleanup-quality-dashboard-dupes.sh
+++ b/.agents/scripts/cleanup-quality-dashboard-dupes.sh
@@ -1,0 +1,79 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2025-2026 Marcus Quinn
+# =============================================================================
+# cleanup-quality-dashboard-dupes.sh — ONE-SHOT cleanup for GH#21830
+# =============================================================================
+# Closes the three duplicate "Code Audit Routines" dashboard issues that
+# accumulated due to the fail-open label-search bug (t3074).
+#
+# Survivor: #21670 (most recent, latest stats)
+# Closed:   #2632, #20409, #20845
+#
+# Usage: bash .agents/scripts/cleanup-quality-dashboard-dupes.sh
+#        REPO_SLUG=owner/repo bash .agents/scripts/cleanup-quality-dashboard-dupes.sh
+#
+# This script is idempotent — re-running it on already-closed issues is safe.
+# =============================================================================
+
+set -euo pipefail
+
+REPO_SLUG="${REPO_SLUG:-marcusquinn/aidevops}"
+SURVIVOR=21670
+# shellcheck disable=SC2206
+DUPES=(2632 20409 20845)
+
+if [[ -t 1 ]]; then
+	GREEN=$'\033[0;32m'
+	RED=$'\033[0;31m'
+	NC=$'\033[0m'
+else
+	GREEN="" RED="" NC=""
+fi
+
+_close_dupe() {
+	local num="$1"
+	local repo_slug="$2"
+	local survivor="$3"
+
+	# Check current state — skip if already closed
+	local state
+	state=$(gh issue view "$num" --repo "$repo_slug" --json state --jq '.state' 2>/dev/null || echo "")
+	if [[ "${state,,}" == "closed" ]]; then
+		printf '  %sSKIP%s #%s (already closed)\n' "$GREEN" "$NC" "$num"
+		return 0
+	fi
+
+	local close_body
+	close_body="> Auto-closed: superseded by #${survivor} as duplicate quality dashboard (t3074 one-time cleanup)."
+	gh issue comment "$num" --repo "$repo_slug" --body "$close_body" 2>/dev/null || true
+	gh issue close "$num" --repo "$repo_slug" 2>/dev/null || {
+		printf '  %sFAIL%s #%s — could not close\n' "$RED" "$NC" "$num"
+		return 1
+	}
+	printf '  %sCLOSED%s #%s\n' "$GREEN" "$NC" "$num"
+	return 0
+}
+
+main() {
+	local repo_slug="$1"
+	local survivor="$2"
+	local failed=0
+
+	printf 'Closing duplicate quality dashboards in %s (survivor: #%s)\n' "$repo_slug" "$survivor"
+
+	local num
+	for num in "${DUPES[@]}"; do
+		_close_dupe "$num" "$repo_slug" "$survivor" || failed=$((failed + 1))
+	done
+
+	if [[ "$failed" -gt 0 ]]; then
+		printf '%sFailed to close %d issue(s).%s\n' "$RED" "$failed" "$NC"
+		return 1
+	fi
+
+	printf '%sDone. %d duplicates closed.%s\n' "$GREEN" "${#DUPES[@]}" "$NC"
+	return 0
+}
+
+main "$REPO_SLUG" "$SURVIVOR"

--- a/.agents/scripts/runtime-audit-rules/counter-trend-delta.sh
+++ b/.agents/scripts/runtime-audit-rules/counter-trend-delta.sh
@@ -79,9 +79,7 @@ runtime_audit_check() {
 	first_counter=$(printf '%s\n' "$regressions" | head -1 | cut -f1)
 
 	local title="runtime-audit: counter regression detected (\`${first_counter}\`)"
-	# Build body via temp file — avoids heredoc-inside-$() which breaks Bash 3.2
-	# (single quotes in heredoc content confuse the Bash 3.2 $() nesting parser).
-	# GH#21782: pre-existing Bash 3.2 compat fix bundled with PyYAML/sed PR.
+	# Bash 3.2 compat: heredoc inside $() fails parse; write to tmp file instead.
 	local _body_tmp
 	_body_tmp=$(mktemp)
 	cat >"$_body_tmp" <<MARKDOWN
@@ -97,7 +95,7 @@ Counter source: \`${stats_file/#$HOME/\~}\`
 
 ## Why
 
-Sharp counter regressions are the supervisor’s structural blind spot — they appear in operational state long before they manifest as failed issues. Each regression cited above warrants investigation: a 3x-or-greater jump on a counter usually means a config change, a new bottleneck, or a regression in a recently-deployed script.
+Sharp counter regressions are the supervisor's structural blind spot — they appear in operational state long before they manifest as failed issues. Each regression cited above warrants investigation: a 3x-or-greater jump on a counter usually means a config change, a new bottleneck, or a regression in a recently-deployed script.
 
 ## How
 

--- a/.agents/scripts/stats-quality-sweep-issues.sh
+++ b/.agents/scripts/stats-quality-sweep-issues.sh
@@ -38,15 +38,126 @@ fi
 # --- Functions ---
 
 #######################################
-# Ensure persistent quality review issue exists for a repo
+# Search for an open quality dashboard issue by labels.
 #
-# Creates or finds the "Daily Code Quality Review" issue. Uses labels
-# "quality-review" + "persistent" for dedup. Pins the issue.
+# Fail-closed: returns 1 on any gh API error so the caller aborts
+# instead of falling through to creation (the original fail-open bug).
+#
+# Arguments:
+#   $1 - repo_slug
+#   $2 - quality-review label name
+#   $3 - persistent label name
+# Output: issue number to stdout (empty if not found)
+# Returns: 0 on success (even if not found), 1 on API error
+#######################################
+_quality_issue_label_search() {
+	local repo_slug="$1"
+	local lbl_review="$2"
+	local lbl_persist="$3"
+
+	local raw_output exit_code
+	raw_output=$(gh issue list --repo "$repo_slug" \
+		--label "$lbl_review" --label "$lbl_persist" \
+		--state open --json number \
+		--jq '.[0].number // empty' 2>&1)
+	exit_code=$?
+
+	if [[ "$exit_code" -ne 0 ]]; then
+		echo "[stats] Quality sweep: label search API error (exit ${exit_code}): ${raw_output}" >>"${LOGFILE:-/dev/null}"
+		return 1
+	fi
+
+	printf '%s' "$raw_output"
+	return 0
+}
+
+#######################################
+# Search for an open quality dashboard issue by title prefix.
+#
+# Used as a fallback when label search returns empty but succeeds.
+# Labels can be stripped by reconcilers; the title prefix is invariant.
+# Fail-closed: returns 1 on any gh API error.
+#
+# Arguments:
+#   $1 - repo_slug
+# Output: issue number to stdout (empty if not found)
+# Returns: 0 on success (even if not found), 1 on API error
+#######################################
+_quality_issue_title_search() {
+	local repo_slug="$1"
+
+	local raw_output exit_code
+	raw_output=$(gh issue list --repo "$repo_slug" \
+		--state open --search "\"Code Audit Routines\" in:title" \
+		--json number --jq '.[0].number // empty' 2>&1)
+	exit_code=$?
+
+	if [[ "$exit_code" -ne 0 ]]; then
+		echo "[stats] Quality sweep: title search API error (exit ${exit_code}): ${raw_output}" >>"${LOGFILE:-/dev/null}"
+		return 1
+	fi
+
+	printf '%s' "$raw_output"
+	return 0
+}
+
+#######################################
+# Close all open "Code Audit Routines" sibling issues except the survivor.
+#
+# Self-healing sweep: called post-create to close any duplicates that
+# accumulated during prior label-search failures. Models on
+# pulse-merge.sh::_close_superseded_prs — close with comment, best-effort.
+#
+# Arguments:
+#   $1 - repo_slug
+#   $2 - survivor_issue_number (the one to keep open)
+# Returns: 0 always (best-effort, never blocks the caller)
+#######################################
+_quality_issue_close_duplicates() {
+	local repo_slug="$1"
+	local survivor="$2"
+
+	local siblings_json
+	siblings_json=$(gh issue list --repo "$repo_slug" \
+		--state open --search "\"Code Audit Routines\" in:title" \
+		--json number --jq '[.[].number]' 2>/dev/null) || return 0
+
+	[[ -z "$siblings_json" ]] && return 0
+
+	local num
+	while IFS= read -r num; do
+		[[ -z "$num" ]] && continue
+		[[ "$num" == "$survivor" ]] && continue
+		local close_body="> Auto-closed: superseded by #${survivor} as duplicate quality dashboard."
+		gh issue comment "$num" --repo "$repo_slug" --body "$close_body" 2>/dev/null || true
+		gh issue close "$num" --repo "$repo_slug" 2>/dev/null || true
+		echo "[stats] Quality sweep: closed duplicate dashboard #${num} (survivor: #${survivor})" >>"${LOGFILE:-/dev/null}"
+	done < <(printf '%s\n' "$siblings_json" | jq -r '.[]' 2>/dev/null)
+
+	return 0
+}
+
+#######################################
+# Ensure persistent quality review issue exists for a repo.
+#
+# Creates or finds the "Code Audit Routines" dashboard issue.
+# Dedup strategy (fail-closed, t3074):
+#   1. Try per-machine cache (fast path).
+#   2. Validate cache via gh issue view — only invalidate on definitive
+#      non-OPEN state; transient API errors keep the cache to prevent
+#      false-create.
+#   3. Label search (fail-closed) — abort entire sweep on API error.
+#   4. Title-prefix fallback — runs only when label search returns empty.
+#      Abort if this API call also fails.
+#   5. Create only when BOTH searches confirmed zero matches.
+#   6. Post-create: run _quality_issue_close_duplicates to reclaim any
+#      siblings created during prior failures.
 #
 # Arguments:
 #   $1 - repo slug
 # Output: issue number to stdout
-# Returns: 0 on success, 1 if issue could not be created/found
+# Returns: 0 on success, 1 if issue could not be found/created or if
+#          any dedup API call fails (fail-closed)
 #######################################
 _ensure_quality_issue() {
 	local repo_slug="$1"
@@ -65,25 +176,42 @@ _ensure_quality_issue() {
 		issue_number=$(cat "$cache_file" 2>/dev/null || echo "")
 	fi
 
-	# Validate cached issue is still open
+	# Validate cached issue — only invalidate on a definitive non-OPEN
+	# response. Transient API failures (non-zero exit, empty state) keep
+	# the cache to avoid triggering the creation path erroneously.
 	if [[ -n "$issue_number" ]]; then
-		local state
-		state=$(gh issue view "$issue_number" --repo "$repo_slug" --json state --jq '.state' 2>/dev/null || echo "")
-		if [[ "$state" != "OPEN" ]]; then
+		local state gh_exit
+		state=$(gh issue view "$issue_number" --repo "$repo_slug" --json state --jq '.state' 2>/dev/null)
+		gh_exit=$?
+		if [[ "$gh_exit" -eq 0 && -n "$state" && "$state" != "OPEN" ]]; then
+			# Definitively closed/merged — invalidate cache
 			issue_number=""
 			rm -f "$cache_file" 2>/dev/null || true
 		fi
+		# Transient failure: keep cache and proceed as if valid
 	fi
 
-	# Search by labels
+	# Label search — fail-closed: abort sweep on any API error
 	if [[ -z "$issue_number" ]]; then
-		issue_number=$(gh issue list --repo "$repo_slug" \
-			--label "$lbl_review" --label "$lbl_persist" \
-			--state open --json number \
-			--jq '.[0].number // empty' 2>/dev/null || echo "")
+		local label_result
+		label_result=$(_quality_issue_label_search "$repo_slug" "$lbl_review" "$lbl_persist") || {
+			echo "[stats] Quality sweep: aborting — label search API failure for ${repo_slug}" >>"${LOGFILE:-/dev/null}"
+			return 1
+		}
+		issue_number="$label_result"
 	fi
 
-	# Create if missing
+	# Title-prefix fallback — only runs when label search returned empty
+	if [[ -z "$issue_number" ]]; then
+		local title_result
+		title_result=$(_quality_issue_title_search "$repo_slug") || {
+			echo "[stats] Quality sweep: aborting — title search API failure for ${repo_slug}" >>"${LOGFILE:-/dev/null}"
+			return 1
+		}
+		issue_number="$title_result"
+	fi
+
+	# Create only when BOTH searches confirmed zero open matches
 	if [[ -z "$issue_number" ]]; then
 		# Ensure labels exist
 		gh label create "$lbl_review" --repo "$repo_slug" --color "7057FF" \
@@ -104,7 +232,7 @@ _ensure_quality_issue() {
 			--label "$lbl_review" --label "$lbl_persist" --label "$lbl_source" 2>/dev/null | grep -oE '[0-9]+$' || echo "")
 
 		if [[ -z "$issue_number" ]]; then
-			echo "[stats] Quality sweep: could not create issue for ${repo_slug}" >>"$LOGFILE"
+			echo "[stats] Quality sweep: could not create issue for ${repo_slug}" >>"${LOGFILE:-/dev/null}"
 			return 1
 		fi
 
@@ -120,10 +248,13 @@ _ensure_quality_issue() {
 				}" >/dev/null 2>&1 || true
 		fi
 
-		echo "[stats] Quality sweep: created and pinned issue #${issue_number} in ${repo_slug}" >>"$LOGFILE"
+		echo "[stats] Quality sweep: created and pinned issue #${issue_number} in ${repo_slug}" >>"${LOGFILE:-/dev/null}"
+
+		# Post-create defensive sweep: close any siblings from prior failures
+		_quality_issue_close_duplicates "$repo_slug" "$issue_number"
 	fi
 
-	# Cache
+	# Cache the winner
 	echo "$issue_number" >"$cache_file"
 	echo "$issue_number"
 	return 0

--- a/.agents/scripts/tests/test-stats-quality-sweep-issues.sh
+++ b/.agents/scripts/tests/test-stats-quality-sweep-issues.sh
@@ -1,0 +1,347 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2025-2026 Marcus Quinn
+#
+# test-stats-quality-sweep-issues.sh — t3074 regression guard.
+#
+# Guards the three changes in _ensure_quality_issue() that fix the
+# cross-runner duplicate "Code Audit Routines" dashboard bug:
+#
+#   (a) label-search-fails-do-not-create:
+#       When gh issue list exits nonzero, _ensure_quality_issue returns 1
+#       and does NOT call gh_create_issue.
+#
+#   (b) title-prefix-fallback-finds-existing:
+#       When label search returns empty (but succeeds), a title-prefix
+#       search runs. If it finds an issue, no new issue is created.
+#
+#   (c) post-create-sweep-closes-siblings:
+#       After a new dashboard is created, _quality_issue_close_duplicates
+#       closes all open sibling issues except the new one.
+#
+# Stub strategy: define `gh`, `gh_create_issue`, `jq` as shell functions
+# AFTER sourcing the library. Shell functions take precedence over PATH
+# binaries. Each test case redefines the stubs to its scenario.
+#
+# Cross-references: GH#21830 / t3074 (fix), GH#10308 (prior class-fix).
+
+set -uo pipefail
+
+SCRIPT_DIR_TEST="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)" || exit 1
+SCRIPTS_DIR="$(cd "${SCRIPT_DIR_TEST}/.." && pwd)" || exit 1
+
+if [[ -t 1 ]]; then
+	TEST_GREEN=$'\033[0;32m'
+	TEST_RED=$'\033[0;31m'
+	TEST_BLUE=$'\033[0;34m'
+	TEST_NC=$'\033[0m'
+else
+	TEST_GREEN="" TEST_RED="" TEST_BLUE="" TEST_NC=""
+fi
+
+TESTS_RUN=0
+TESTS_FAILED=0
+
+pass() {
+	TESTS_RUN=$((TESTS_RUN + 1))
+	printf '  %sPASS%s %s\n' "$TEST_GREEN" "$TEST_NC" "$1"
+	return 0
+}
+
+fail() {
+	TESTS_RUN=$((TESTS_RUN + 1))
+	TESTS_FAILED=$((TESTS_FAILED + 1))
+	printf '  %sFAIL%s %s\n' "$TEST_RED" "$TEST_NC" "$1"
+	if [[ -n "${2:-}" ]]; then
+		printf '       %s\n' "$2"
+	fi
+	return 0
+}
+
+# =============================================================================
+# Sandbox setup
+# =============================================================================
+TMP=$(mktemp -d -t t3074.XXXXXX)
+trap 'rm -rf "$TMP"' EXIT
+
+GH_CALLS="${TMP}/gh_calls.log"
+CREATE_CALLS="${TMP}/create_calls.log"
+CLOSE_CALLS="${TMP}/close_calls.log"
+LOGFILE="${TMP}/test.log"
+export LOGFILE
+
+# =============================================================================
+# Source the library with quiet stubs
+# =============================================================================
+print_info() { :; }
+print_warning() { :; }
+print_error() { :; }
+print_success() { :; }
+log_verbose() { :; }
+export -f print_info print_warning print_error print_success log_verbose
+
+# gh_create_issue is a wrapper defined in shared-constants.sh; stub it directly
+gh_create_issue() {
+	echo "gh_create_issue $*" >>"$CREATE_CALLS"
+	# Simulate successful creation returning a URL with issue number
+	printf 'https://github.com/test/repo/issues/999\n'
+	return 0
+}
+export -f gh_create_issue
+
+# gh_issue_edit_safe stub
+gh_issue_edit_safe() { return 0; }
+export -f gh_issue_edit_safe
+
+# Suppress stats-quality-sweep-coverage.sh dependency functions
+_compute_bot_coverage() { printf 'no coverage'; return 0; }
+_compute_badge_indicator() { printf 'OK'; return 0; }
+export -f _compute_bot_coverage _compute_badge_indicator
+
+# shellcheck source=../stats-quality-sweep-issues.sh
+source "${SCRIPTS_DIR}/stats-quality-sweep-issues.sh" 2>/dev/null || {
+	printf '%sFATAL%s could not source stats-quality-sweep-issues.sh\n' "$TEST_RED" "$TEST_NC"
+	exit 1
+}
+
+# =============================================================================
+# Test (a): label search API failure → abort, no creation
+# =============================================================================
+printf '\n%s[a] label-search-fails-do-not-create%s\n' "$TEST_BLUE" "$TEST_NC"
+
+# Clear logs
+true >"$CREATE_CALLS"
+true >"$GH_CALLS"
+
+# Stub: gh issue list returns nonzero (simulates API error).
+# No cache file — forces the function to go through the label search path.
+gh() {
+	echo "gh $*" >>"$GH_CALLS"
+	case "$*" in
+	*"issue list"*)
+		# Simulate transient API failure
+		printf 'error: API rate limit exceeded\n' >&2
+		return 1
+		;;
+	*"label create"*)
+		return 0
+		;;
+	esac
+	return 0
+}
+export -f gh
+
+jq() { command jq "$@"; return 0; }
+export -f jq
+
+# Override HOME so cache file is isolated; ensure no stale cache exists
+_original_home="$HOME"
+export HOME="$TMP"
+mkdir -p "${TMP}/.aidevops/logs"
+rm -f "${TMP}/.aidevops/logs/quality-issue-test-repo"
+
+result=$(_ensure_quality_issue "test/repo" 2>/dev/null)
+rc=$?
+export HOME="$_original_home"
+
+if [[ "$rc" -ne 0 ]]; then
+	pass "label-search-fails-returns-1"
+else
+	fail "label-search-fails-returns-1" "expected return 1, got $rc"
+fi
+
+if [[ ! -s "$CREATE_CALLS" ]]; then
+	pass "label-search-fails-no-create"
+else
+	fail "label-search-fails-no-create" "gh_create_issue was called: $(cat "$CREATE_CALLS")"
+fi
+
+# =============================================================================
+# Test (b): title-prefix fallback finds existing issue → no creation
+# =============================================================================
+printf '\n%s[b] title-prefix-fallback-finds-existing%s\n' "$TEST_BLUE" "$TEST_NC"
+
+true >"$CREATE_CALLS"
+true >"$GH_CALLS"
+
+# Stub: label search succeeds but returns empty; title search finds #456
+gh() {
+	echo "gh $*" >>"$GH_CALLS"
+	case "$*" in
+	*"issue view"*)
+		printf 'OPEN\n'
+		return 0
+		;;
+	*"issue list"*"quality-review"*)
+		# Label search: returns empty (no labels found) — exit 0
+		printf ''
+		return 0
+		;;
+	*"issue list"*"Code Audit Routines"*)
+		# Title-prefix fallback: finds existing issue 456
+		printf '456\n'
+		return 0
+		;;
+	*"label create"*)
+		return 0
+		;;
+	esac
+	return 0
+}
+export -f gh
+
+jq() { command jq "$@"; return 0; }
+export -f jq
+
+export HOME="$TMP"
+rm -f "${TMP}/.aidevops/logs/quality-issue-test-repo"
+
+result=$(_ensure_quality_issue "test/repo" 2>/dev/null)
+rc=$?
+export HOME="$_original_home"
+
+if [[ "$rc" -eq 0 && "$result" == "456" ]]; then
+	pass "title-fallback-returns-existing-number"
+else
+	fail "title-fallback-returns-existing-number" "expected 0/456, got rc=$rc result='$result'"
+fi
+
+if [[ ! -s "$CREATE_CALLS" ]]; then
+	pass "title-fallback-no-create"
+else
+	fail "title-fallback-no-create" "gh_create_issue was called: $(cat "$CREATE_CALLS")"
+fi
+
+# Verify title search was called (check gh log contains "Code Audit Routines")
+if grep -q "Code Audit Routines" "$GH_CALLS" 2>/dev/null; then
+	pass "title-search-was-invoked"
+else
+	fail "title-search-was-invoked" "title search not found in gh calls: $(cat "$GH_CALLS")"
+fi
+
+# =============================================================================
+# Test (c): post-create defensive sweep closes siblings
+# =============================================================================
+printf '\n%s[c] post-create-sweep-closes-siblings%s\n' "$TEST_BLUE" "$TEST_NC"
+
+true >"$CREATE_CALLS"
+true >"$GH_CALLS"
+true >"$CLOSE_CALLS"
+
+# Track what gets closed
+CLOSED_NUMS="${TMP}/closed_nums.log"
+true >"$CLOSED_NUMS"
+
+# Stub: both searches return empty → create fires; sibling search finds 2 extras
+gh() {
+	echo "gh $*" >>"$GH_CALLS"
+	case "$*" in
+	*"issue view"*"--json state"*)
+		printf 'OPEN\n'
+		return 0
+		;;
+	*"issue view"*"--json id"*)
+		printf '{"id":"node_abc"}\n' | command jq -r '.id'
+		return 0
+		;;
+	*"issue list"*"quality-review"*)
+		# Label search: empty
+		printf ''
+		return 0
+		;;
+	*"issue list"*"Code Audit Routines"*".[0].number"*)
+		# Title-prefix search (--jq '.[0].number // empty'): return empty
+		# so the create path executes
+		printf ''
+		return 0
+		;;
+	*"issue list"*"Code Audit Routines"*"[.[].number]"*)
+		# Sibling search in _quality_issue_close_duplicates (--jq '[.[].number]'):
+		# return a JSON array containing the new issue (999) plus two siblings
+		printf '[100,200,999]\n'
+		return 0
+		;;
+	*"issue comment"*)
+		local num
+		num=$(printf '%s' "$*" | grep -oE '[0-9]+' | head -1)
+		echo "comment:$num" >>"$CLOSE_CALLS"
+		return 0
+		;;
+	*"issue close"*)
+		local num
+		num=$(printf '%s' "$*" | grep -oE '[0-9]+' | head -1)
+		echo "close:$num" >>"$CLOSE_CALLS"
+		return 0
+		;;
+	*"label create"*)
+		return 0
+		;;
+	*"api graphql"*)
+		return 0
+		;;
+	esac
+	return 0
+}
+export -f gh
+
+jq() {
+	if [[ "${1:-}" == "-r" && "${2:-}" == ".[]" ]]; then
+		# Parse the JSON array from siblings search
+		command jq -r '.[]' 2>/dev/null || printf '100\n200\n999\n'
+		return 0
+	fi
+	command jq "$@"
+	return 0
+}
+export -f jq
+
+export HOME="$TMP"
+rm -f "${TMP}/.aidevops/logs/quality-issue-test-repo"
+# Also clear the gh-signature-helper output
+GH_SIG_CALLED="${TMP}/sig_called"
+true >"$GH_SIG_CALLED"
+
+# Stub gh-signature-helper.sh
+# shellcheck disable=SC2034
+_orig_home_sig="$HOME"
+
+result=$(_ensure_quality_issue "test/repo" 2>/dev/null)
+rc=$?
+export HOME="$_original_home"
+
+if [[ "$rc" -eq 0 ]]; then
+	pass "post-create-sweep-create-succeeded"
+else
+	fail "post-create-sweep-create-succeeded" "expected rc=0, got $rc"
+fi
+
+# Check that close was called for siblings (100, 200) but NOT for survivor (999)
+if grep -q "close:100" "$CLOSE_CALLS" 2>/dev/null; then
+	pass "post-create-sweep-closed-sibling-100"
+else
+	fail "post-create-sweep-closed-sibling-100" "close:100 not found in: $(cat "$CLOSE_CALLS" 2>/dev/null)"
+fi
+
+if grep -q "close:200" "$CLOSE_CALLS" 2>/dev/null; then
+	pass "post-create-sweep-closed-sibling-200"
+else
+	fail "post-create-sweep-closed-sibling-200" "close:200 not found in: $(cat "$CLOSE_CALLS" 2>/dev/null)"
+fi
+
+if ! grep -q "close:999" "$CLOSE_CALLS" 2>/dev/null; then
+	pass "post-create-sweep-preserved-survivor"
+else
+	fail "post-create-sweep-preserved-survivor" "survivor 999 was incorrectly closed"
+fi
+
+# =============================================================================
+# Summary
+# =============================================================================
+printf '\n'
+if [[ "$TESTS_FAILED" -eq 0 ]]; then
+	printf '%sAll %d tests passed.%s\n' "$TEST_GREEN" "$TESTS_RUN" "$TEST_NC"
+	exit 0
+else
+	printf '%s%d of %d tests FAILED.%s\n' "$TEST_RED" "$TESTS_FAILED" "$TESTS_RUN" "$TEST_NC"
+	exit 1
+fi

